libgnucash/engine/Account.cpp

```diff
@@ -4142,14 +4205,24 @@ xaccAccountGetIsOpeningBalance (const Account *acc)
 {
     if (GET_PRIVATE(acc)->type != ACCT_TYPE_EQUITY)
         return false;
-    return g_strcmp0(get_kvp_string_tag(acc, "equity-type"), "opening-balance") == 0;
+    auto priv = GET_PRIVATE(acc);
+    if (priv->equity_type == TriState::Unset)
+    {
+        auto equity_type = get_kvp_string_tag (acc, "equity-type");
+        priv->equity_type = g_strcmp0 (equity_type, "true") ?
```

Comment:
oops error should be `priv->equity_type = g_strcmp0 (equity_type, "opening-balance") ?`